compiler/rustc_ast/src/token.rs

```diff
@@ -688,13 +688,13 @@ pub enum NonterminalKind {
     Item,
     Block,
     Stmt,
-    Pat2015 {
-        /// Keep track of whether the user used `:pat2015` or `:pat` and we inferred it from the
+    PatParam {
+        /// Keep track of whether the user used `:pat_param` or `:pat` and we inferred it from the
         /// edition of the span. This is used for diagnostics.
         inferred: bool,
     },
     Pat2021 {
```

Comment:
I would rather leave it as `Pat2021` or name it `PatWithOr` or something. Otherwise, I think the code would confusing given that it handles all editions.